packages/core/chrome/core-chrome-browser-internal/src/ui/project/navigation.tsx

```diff
@@ -17,23 +17,52 @@ interface Props {
   isSideNavCollapsed$: Observable<boolean>;
 }
 
+const PANEL_WIDTH = 290;
+
 export const ProjectNavigation: FC<PropsWithChildren<Props>> = ({
   children,
   isSideNavCollapsed$,
   toggleSideNav,
 }) => {
   const isCollapsed = useObservable(isSideNavCollapsed$, false);
+  const [collapsibleNavOffset, setCollapsibleNavOffset] = useState(0);
+  const clipPathWidth = collapsibleNavOffset + PANEL_WIDTH;
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout | undefined;
+    const observer = new MutationObserver(() => {
+      clearTimeout(timeoutId);
+
+      // Debounce the CSS variable change to avoid unnecessary re-renders
+      timeoutId = setTimeout(() => {
+        const offSet = getComputedStyle(document.documentElement).getPropertyValue(
+          '--euiCollapsibleNavOffset'
+        );
+        setCollapsibleNavOffset(parseFloat(offSet));
+      }, 50);
+    });
+
+    observer.observe(document.documentElement, {
+      attributes: true,
+      attributeFilter: ['style'],
+      subtree: true,
+      childList: true,
+    });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
 
   return (
     <EuiCollapsibleNavBeta
       data-test-subj="projectLayoutSideNav"
       isCollapsed={isCollapsed}
       onCollapseToggle={toggleSideNav}
-      css={
-        isCollapsed
-          ? undefined
-          : { overflow: 'visible', clipPath: 'polygon(0 0, 300% 0, 300% 100%, 0 100%)' }
-      }
+      css={{
+        overflow: 'visible',
+        clipPath: `polygon(0 0, ${clipPathWidth}px 0, ${clipPathWidth}px 100%, 0 100%)`,
```

Comment:
Hm... can we try css-only solution here? 
Wouldn't this work? 

```
clipPath: `polygon(0 0, calc(var(--euiCollapsibleNavOffset) + ${PANEL_WIDTH}px) 0, calc(var(--euiCollapsibleNavOffset) + ${PANEL_WIDTH}px) 100%, 0 100%)`,
```

Reply:
:hide:

Reply:
Yep, not sure which planet I was on! Fixed https://github.com/elastic/kibana/pull/194268/commits/995ebaf8663f1d0befd970bbcc104979389ddd72